src/ff1.rs

```diff
@@ -127,50 +127,46 @@ impl Radix {
     }
 }
```

Comment:
Suggestion for simpler `Radix::from_u32` (above this):
```rust
    fn from_u32(radix: u32) -> Result<Self, InvalidRadix> {
        // radix must be in range [2..=2^16]
        if !(2..=(1 << 16)).contains(&radix) {
            return Err(InvalidRadix(radix));
        }

        Ok(if radix.count_ones() == 1 {
            let lg_radix: u32 = 31 - radix.leading_zeros();
            Radix::PowerTwo {
                radix,
                min_len: cmp::max((MIN_RADIX_2_NS_LEN + lg_radix - 1) / lg_radix, MIN_NS_LEN),
                lg_radix,
            }
        } else {
            use libm::{ceil, log10};
            let min_len = ceil(LOG_10_MIN_NS_DOMAIN_SIZE / log10(f64::from(radix))) as u32;
            Radix::Any { radix, min_len }
        })
    }
```

Reply:
Ah yes, we can indeed use `u32::count_ones` and `u32::leading_zeros` now (they were added in 1.32, seven months after I wrote `fpe 0.1` for Sapling). I also bumped MSRV to 1.56 for this upcoming release, so we can use `edition = "2021"` as well. I'll make both of these changes in a separate PR (as it's unrelated to the changes in this one).